contracts/starknet/jug.cairo

```diff
@@ -0,0 +1,433 @@
+// // SPDX-License-Identifier: AGPL-3.0-or-later
+
+// /// Jug.sol -- Dai Lending Rate
+
+// // Copyright (C) 2018 Rain <rainbreak@riseup.net>
+// // Copyright (C) 2022 Dai Foundation
+// //
+// // This program is free software: you can redistribute it and/or modify
+// // it under the terms of the GNU Affero General Public License as published by
+// // the Free Software Foundation, either version 3 of the License, or
+// // (at your option) any later version.
+// //
+// // This program is distributed in the hope that it will be useful,
+// // but WITHOUT ANY WARRANTY; without even the implied warranty of
+// // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// // GNU Affero General Public License for more details.
+// //
+// // You should have received a copy of the GNU Affero General Public License
+// // along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// pragma solidity ^0.8.13;
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from contracts.starknet.safe_math import Int256, add, _felt_to_uint, div_rem, mul, div, _sub
+from contracts.starknet.assertions import _ge_0, assert_0
+from starkware.starknet.common.syscalls import (
+    get_caller_address,
+    get_contract_address,
+    get_block_timestamp,
+)
+
+from starkware.cairo.common.uint256 import Uint256, uint256_le, uint256_neg, uint256_eq
+
+// interface VatLike {
```

Comment:
Add a link to relevant solidity file. I.e.: https://github.com/makerdao/xdomain-dss/blob/...